src/main/java/ssagit/parser/Parser.java

```diff
@@ -0,0 +1,132 @@
+package ssagit.parser;
+
+import ssagit.datevalidator.DateValidator;
+import ssagit.taskclass.DeadlineTask;
+import ssagit.taskclass.EventTask;
+import ssagit.taskclass.Task;
+import ssagit.ui.ConsoleUI;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Parser {
+    ConsoleUI ui;
+    boolean isBye;
+    DateValidator validator;
+
+    public Parser(ConsoleUI ui, DateValidator validator) {
+        this.ui = ui;
+        this.isBye = false;
+        this.validator = validator;
+    }
+
+    /**
+     * Exception class for missing todoTask descriptor.
+     */
+    static class MissingTodoDescriptorException extends Exception {
+        public MissingTodoDescriptorException(String errorMessage) {
+            super(errorMessage);
+        }
+    }
+
+    /**
+     * Exception class for unknown input parameters.
+     */
+    static class UnknownInputParamException extends Exception {
+        public UnknownInputParamException(String errorMessage) {
+            super(errorMessage);
+        }
+    }
+
+    public boolean getIsBye() {
+        return this.isBye;
+    }
+
+    /**
+     * Parses lines of user input and outputs corresponding command.
+     * @param tasks List of tasks from file.
+     * @param taskIterator Integer to count number of tasks at a time.
+     */
+    public void parseInput(Task[] tasks, int taskIterator) {
+        String input = ui.nextLine();
+        String inputArr[] = input.split(" ", 2);
+
+        try {
+            switch (inputArr[0]) {
+            case "bye":
+                this.isBye = true;
+                break;
+            case "list":
+                ui.list(tasks);
+                break;
+            case "done":
+                int taskNum = Integer.parseInt(inputArr[1]) - 1;
+                tasks[taskNum].markDone();
+                ui.markDone(tasks[taskNum].toFormattedString());
+                break;
+            case "todo":
+            case "event":
+            case "deadline":
```

Comment:
This doesn't seem like good practice to me, should handle the todo and event in their respective cases I feel